routing/graph.go

```diff
@@ -33,6 +33,59 @@ type ReadOnlyGraph interface {
 	Graph
 }
 
+type GraphSessionConstructor interface {
+	NewSession() (GraphSession, error)
+}
+
+type GraphSession interface {
+	Graph() routingGraph
```

Comment:
How can this be implemented outside the package if it returns a private interface? 

Alternatively, it can embed a public version of `routingGraph`, and then add the `Close` method.